src/components/HTMLEngineProvider/HTMLRenderers/ImageRenderer.js

```diff
@@ -49,18 +49,28 @@ const ImageRenderer = (props) => {
             isAuthTokenRequired={isAttachment}
             originalFileName={originalFileName}
         >
-            {({show}) => (
-                <TouchableOpacity
-                    style={styles.noOutline}
-                    onPress={show}
-                >
-                    <ThumbnailImage
-                        previewSourceURL={previewSource}
-                        style={styles.webViewStyles.tagStyles.img}
-                        isAuthTokenRequired={isAttachment}
-                    />
-                </TouchableOpacity>
-            )}
+            {({show}) => {
+                const openModal = () => {
+                    // Blur the list item which got tapped so that it won't get refocused on modal close.
+                    if (document.activeElement) {
+                        document.activeElement.blur();
+                    }
+                    show();
+                };
+
+                return (
+                    <TouchableOpacity
+                        style={styles.noOutline}
+                        onPress={openModal}
+                    >
+                        <ThumbnailImage
+                            previewSourceURL={previewSource}
+                            style={styles.webViewStyles.tagStyles.img}
+                            isAuthTokenRequired={isAttachment}
+                        />
+                    </TouchableOpacity>
+                );
+            }}
```

Comment:
Let's make a thin wrapper that resets the focus on click before firing the onPress callback. 

This wrapper will encapsulate the 

```js
  <TouchableOpacity
                        style={styles.noOutline}
                        onPress={openModal}
                    >
```

Reply:
@parasharrajat You mean something like this? In that case it is better to track the ref and call blur on the item itself so that it could be used even in native code.

```jsx
class TouchableWithoutFocus extends React.Component {
    constructor() {
        super();
        this.onPress = this.onPress.bind(this);
    }

    onPress() {
        if (!this.props.onPress) {
            return;
        }
        this.props.onPress();
        this.inputRef.blur();
    }

    render() {
        return (
            <TouchableOpacity onPress={this.onPress} ref={e => this.inputRef = e} style={this.props.styles}>
                {this.props.children}
            </TouchableOpacity>
        );
    }
}
```

Reply:
Sounds good.